### PR TITLE
Doc - Add dynamic TCE version

### LINKED ISSUE #2875: Add subnote to docs that mention to not use `:latest` when adding a package repository
## Feature Request

Not sure if there are others, but I see at least three places where the latest version of TCE is mentioned:
1. This repo. At the time of creation of this issue, [0.9.1](https://github.com/vmware-tanzu/community-edition/releases/tag/v0.9.1) is labeled as latest
2. The package installation instructions in the docs, which at the time of creation of this issue, also say 0.9.1:
     URL: https://tanzucommunityedition.io/docs/latest/package-management/
```
tanzu package repository add tce-repo \
  --url projects.registry.vmware.com/tce/main:0.9.1
```
3. The tag on the package image on Harbor. 
     This is where I see a discrepancy. On Harbor, main:0.9.1 and main:latest
```
~$ docker images
REPOSITORY                                   TAG                  IMAGE ID       CREATED        SIZE
projects.registry.vmware.com/tce/main        latest               f454986a53e1   N/A            273kB
projects.registry.vmware.com/tce/main        0.9.1                2835b13e3647   N/A            217kB
```

It would be great if the image were also in sync so that using :latest would be an option.
As it is, it is not clear what that latest image relates to, so it cannot be used with confidence.

In fact, the docs could use :latest in the example commands and then have a link to where the user can look up the latest. Otherwise it is not clear to the user if the docs are up to date or if there is something newer, and perhaps not entirely obvious where to look.

### PR DESCRIPTION
**Pull request by kcoriordan**
## What this PR does / why we need it
Add a new parameter setting in `config.yaml ` for **release_latest_no_v** which is the latest TCE version not preceded by 'v', which is then picked up by the new shortcode. 

## Details for the Release Notes (PLEASE PROVIDE)

```none

```

## Which issue(s) this PR fixes

Fixes: #2875

## Describe testing done for PR
Run `hugo server `as per instructions in [readme](https://github.com/vmware-tanzu/community-edition/blob/main/docs/README.md)

### CODE REVIEW

**Comment by dvonthenen** [comment]
I need to update the automation to also replace `release_latest_no_v:` when a new version comes out

**Comment by dvonthenen** [comment]
Please see:
https://github.com/vmware-tanzu/community-edition/pull/2957

**Review by davidvonthenen — APPROVED** [review]

**Comment by dvonthenen** [comment]
Before we merge, we should probably get approval on https://github.com/vmware-tanzu/community-edition/pull/2957 to merge at the same time

**Pull request merged**
